rule-engine/rule-engine-components/src/main/java/org/thingsboard/rule/engine/transform/TbDeleteKeysNode.java

```diff
@@ -0,0 +1,112 @@
+import java.util.regex.Pattern;
+
+@Slf4j
+@RuleNode(
+        type = ComponentType.TRANSFORMATION,
+        name = "delete keys",
+        configClazz = TbDeleteKeysNodeConfiguration.class,
+        nodeDescription = "Removes keys from the msg data or metadata with the specified key names selected in the list",
+        nodeDetails = "Will fetch fields (regex) values specified in list. If specified field (regex) is not part of msg " +
+                "or metadata fields it will be ignored. Returns transformed messages via <code>Success</code> chain",
+        uiResources = {"static/rulenode/rulenode-core-config.js"},
+        configDirective = "tbTransformationNodeDeleteKeysConfig",
+        icon = "remove_circle"
+)
+public class TbDeleteKeysNode implements TbNode {
+
+    TbDeleteKeysNodeConfiguration config;
+    List<Pattern> patternKeys = new ArrayList<>();
+    boolean fromMetadata;
+
+    @Override
+    public void init(TbContext ctx, TbNodeConfiguration configuration) throws TbNodeException {
+        this.config = TbNodeUtils.convert(configuration, TbDeleteKeysNodeConfiguration.class);
+        this.fromMetadata = config.isFromMetadata();
+        config.getKeys().forEach(key -> {
+            this.patternKeys.add(Pattern.compile(key));
+        });
+    }
+
+    @Override
+    public void onMsg(TbContext ctx, TbMsg msg) throws ExecutionException, InterruptedException, TbNodeException {
+        TbMsgMetaData metaData = msg.getMetaData();
+        String msgData = msg.getData();
+        List<String> keysToDelete = new ArrayList<>();
+        if (fromMetadata) {
+            Map<String, String> metaDataMap = metaData.getData();
+            metaDataMap.forEach((keyMetaData, valueMetaData) -> {
+                if (checkKey(keyMetaData)) {
+                    keysToDelete.add(keyMetaData);
+                }
+            });
+            keysToDelete.forEach(key -> metaDataMap.remove(key));
+            metaData = new TbMsgMetaData(metaDataMap);
+        } else {
+            JsonNode dataNode = JacksonUtil.toJsonNode(msgData);
+            if (dataNode.isObject()) {
+                ObjectNode msgDataObject = (ObjectNode) dataNode;
+                dataNode.fields().forEachRemaining(entry -> {
+                    String keyData = entry.getKey();
+                    if (checkKey(keyData)) {
+                        keysToDelete.add(keyData);
+                    }
+                });
+                msgDataObject.remove(keysToDelete);
+                msgData = JacksonUtil.toString(msgDataObject);
+            }
+        }
+        if (keysToDelete.isEmpty()) {
+            ctx.tellSuccess(msg);
+        } else {
+            ctx.tellSuccess(TbMsg.transformMsg(msg, msg.getType(), msg.getOriginator(), metaData, msgData));
+        }
+    }
+
+    boolean checkKey(String key) {
+        Optional<Pattern> currentPattern = patternKeys.stream()
+                .filter(pattern -> pattern.matcher(key).matches())
```

Comment:
@YuriyLytvynchuk, probably the `anyMatch(Predicate<? super T> var1);` method will be better to use here. Review the Stream.class docs.